ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -223,19 +264,20 @@ private static SourceDescriptionWithWarnings describeSource(
     Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription =
         Optional.empty();
     List<QueryOffsetSummary> queryOffsetSummaries = Collections.emptyList();
-    List<String> sourceConstraints = Collections.emptyList();
+    List<String> sourceConstraints = getSourceConstraints(name, ksqlEngine.getMetaStore());
 
     final List<KsqlWarning> warnings = new LinkedList<>();
+    try {
+      topicDescription = Optional.of(
+          serviceContext.getTopicClient().describeTopic(dataSource.getKafkaTopicName())
+      );
+      sourceConstraints = getSourceConstraints(name, ksqlEngine.getMetaStore());
```

Comment:
do we still need this? looks like we're collecting the constraints up on 267 as well

Reply:
Ah nice catch, we don't